zhmc_prometheus_exporter/zhmc_prometheus_exporter.py

```diff
@@ -900,6 +920,35 @@ def expand_global_label_value(
     return str(value)
 
 
+def get_backing_adapter_info(nic):
+    """
+    Return backing adapter and port of the specified NIC.
+
+    Returns:
+      tuple(adapter_name, port_index)
+    """
+
+    session = nic.manager.session
+
+    # Handle vswitch-based NIC (OSA, HS)
+    try:
+        vswitch_uri = nic.get_property('virtual-switch-uri')
+    except KeyError:
```

Comment:
Suggestion: Instead of doing this way, where we are consuming this exception if the nic object doesn't have 'virtual-switch-uri'........ can we check the type of the nic using nic.get_property('type') which gives us "roce"/"cna"/"iqd"/"osd", and based on the type of nic we'll get the adapter name and adapter port

Reply:
If we check the adapter type, we have to know which adapter types have a virtual switch. If tomorrow, a new adapter type emerges, our list can become outdated. If we check for the 'virtual-switch-uri' property, we are prepared for any future adapter type that is virtual-switch based.

So I think we should continue checking for the 'virtual-switch-uri' property. Also, it is considered more "pythonic" to do it that way by some people, see https://realpython.com/python-lbyl-vs-eafp/ for a discussion of the pro's and con's.